documentation/src/main/asciidoc/public/reference/_search-dsl-predicate.adoc

```diff
@@ -1464,6 +1464,53 @@ either on a per-field basis with a call to `.boost(...)` just after `.field(...)
 or for the whole predicate with a call to `.boost(...)`
 after `.circle(...)`/`.boundingBox(...)`/`.polygon(...)`.
 
+[[search-dsl-predicate-knn]]
+== `knn`: K-Nearest Neighbors aka vector search
+
+include::../components/_incubating-warning.adoc[]
+
+The `knn` predicate matches k nearest documents to a query vector based on the distance to the given vector field values.
+Distance is measured based on the vector similarity configured for the given vector field.
+// TODO : vector : links to a vector field info ^^^
+
+.Simple K-Nearest Neighbors search
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/LucenePredicateDslIT.java[tags=knn]
+----
+====
+
+[[search-dsl-predicate-knn-argument-type]]
+=== Expected type of arguments
+
+The `knn` predicate expects arguments to the `matching(...)` method
+to have the same type as the entity property corresponding to the target field.
+
+For example, if an entity property is of a byte array type (`byte[]`),
+`.matching(...)` will expect its argument to be a byte array (`byte[]`) only.
```

Comment:
> If the property is of type List<Byte> and we set up a custom bridge, it will expect a List<Byte>, right? Not byte[]? And there's a test for that?

mmm not really the predicate will only take an array (`KnnPredicateVectorStep`)

Reply:
Ok, then the description is incorrect. The expected type has nothing to do with the entity property type. You could have `byte[]` in the entity and use a `ValueBridge<byte[], float[]>`, and then the KNN predicate expects a `float[]`.